File: Data Structures & Algorithms/daily-temperatures/submission-4.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Deque<Integer> stack = new ArrayDeque<>(temperatures.length);
        int[] res = new int[temperatures.length];
        for (int i = 0; i < temperatures.length; i++) {
            while (!stack.isEmpty() && temperatures[stack.getFirst()] < temperatures[i]) {
                int pos = stack.pop();
                res[pos] = i - pos;
            }
            stack.addFirst(i);
        }
        return res;
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-4.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        if k == 1:
            return nums
        deque = []
        res = []
        for i,e in enumerate(nums):
            while deque and deque[-1] < e:
                deque.pop(-1)
            deque.append(e)
            if i > k - 1 and nums[i - k] == deque[0]:
                deque.pop(0)
            if i >= k - 1:
                res.append(deque[0])
        return res